AK/Time.h

```diff
@@ -354,6 +354,125 @@ class Duration {
     u32 m_nanoseconds { 0 }; // Always less than 1'000'000'000
 };
 
+namespace Detail {
+
+// Common base class for all naive time types. This class and its children are unaware of timezones.
+struct NaiveTime {
+    constexpr NaiveTime(NaiveTime const&) = default;
+    constexpr NaiveTime& operator=(NaiveTime const&) = default;
+
+    [[nodiscard]] timespec to_timespec() const { return offset.to_timespec(); }
+    // Rounds towards -inf.
+    [[nodiscard]] timeval to_timeval() const { return offset.to_timeval(); }
+
+    // We intentionally do not define a comparison operator here to avoid accidentally comparing incompatible time types.
+
+protected:
+    constexpr explicit NaiveTime(Duration offset)
+        : offset(offset)
+    {
+    }
+
+    Duration offset {};
```

Comment:
```suggestion
    Duration m_offset {};
```

I think we only remove the leading `m_` for POD-like types, or whatever those are called.